tests/Composer/ArtifactsPluginTest.php

```diff
@@ -136,15 +138,27 @@ public function testRegistersProviders(): void
         $composer = $this->mockComposerWithDataDir(
             $config,
             null,
-            [$this->mockPackage('foo/bar', 'contao-provider', __DIR__.'/../Fixtures/Composer/provider-data/packages/foo-bar-1.0.0.zip')],
-            ['foo/bar' => '*']
+            [$this->mockPackage(
+                'foo/bar',
+                'contao-provider',
+                '1.0.0',
+                __DIR__.'/../Fixtures/Composer/provider-data/packages/foo-bar-1.0.0.zip'
+            )],
+            ['foo/bar' => true]
         );
 
+        /** @var PackageInterface|MockObject $rootPackage */
+        $rootPackage = $composer->getPackage();
+        $rootPackage
+            ->expects($this->once())
+            ->method('setRepositories')
```

Comment:
Imho this should test wether `with()` got the correct repository, shouldn't it?

Reply:
The definition of *correct* is a bit opaque. Because it's the return value of `$config->getRepositories()`, and by default that value is `null` in the tests because the `$config` object is mocked. However, it is tested in https://github.com/contao/manager-plugin/pull/20/files/fd382981767e6d1c3c1e5c10262062f63eb1fb46#diff-7bcc5fcd6778c9a9e6764ad1dcf9700cR288